cmd/virt-launcher/virt-launcher.go

```diff
@@ -130,27 +134,31 @@ func startDomainEventMonitoring(virtShareDir string, domainConn virtcli.Connecti
 		}
 	}()
 
-	err := notifyclient.StartNotifier(virtShareDir, domainConn, deleteNotificationSent)
+	err := notifyclient.StartNotifier(virtShareDir, domainConn, deleteNotificationSent, vmiUID)
 	if err != nil {
 		panic(err)
 	}
 
 }
 
-func startWatchdogTicker(watchdogFile string, watchdogInterval time.Duration, stopChan chan struct{}) {
+func startWatchdogTicker(watchdogFile string, watchdogInterval time.Duration, stopChan chan struct{}) (done chan struct{}) {
 	err := watchdog.WatchdogFileUpdate(watchdogFile)
 	if err != nil {
 		panic(err)
 	}
 
 	log.Log.Infof("Watchdog file created at %s", watchdogFile)
+	done = make(chan struct{})
 
 	go func() {
+		defer close(done)
 
 		ticker := time.NewTicker(watchdogInterval).C
 		for {
 			select {
 			case <-stopChan:
+				err := os.RemoveAll(watchdogFile)
```

Comment:
Only virt-handler in the processVMCleanup function should ever remove the watchdog file.  virt-launcher creates it. virt-handler cleans it up.  The presence of the watchdog file means that artifact cleanup has not occurred yet.

so, deletion of the watchdog file needs to always be coupled with the local removal of all other artifacts associated with the VMI.

Reply:
thx, reverted.